python-package/lightgbm/compat.py

```diff
@@ -338,6 +326,27 @@ class pa_compute:  # type: ignore
     arrow_is_integer = None
     arrow_is_floating = None
 
+
+"""cffi"""
+try:
+    from pyarrow.cffi import ffi as arrow_cffi
+
+    CFFI_INSTALLED = True
+except ImportError:
+    CFFI_INSTALLED = False
+
+    class arrow_cffi:  # type: ignore
+        """Dummy class for pyarrow.cffi.ffi."""
+
+        CData = None
+        addressof = None
+        cast = None
+        new = None
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        pass
```

Comment:
```suggestion
        def __init__(self, *args: Any, **kwargs: Any):
            pass
```

This indentation means we'd be defining a function `compat.__init__`. I'll fix this in the next commit I push here.

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/6785/commits/e72d5e21521eb23492b689a3a361fa927286b3d3